### PR TITLE
refactor how target projection is determined

### PR DESCRIPTION
**Pull request by asjohnston-asf**
I'm not convinced this is a major improvement, and I'm out until Monday, but I'll leave it here if anyone wants to poke at it or move it forward.

This implementation moves all of the utm-zone-aware math into a single `get_target_epsg_code` function.  A big advantage is that `reproject_to_median_utm` now only knows about the target EPSG code and the individual file EPSG codes (and none of the logic behind them).

There are probably better ways to implement `get_epsg_code` and `get_target_epsg_code`.  They both deserve tests.  They'll be a little difficult to test in this form since they both take file names as inputs.

### CODE REVIEW

File: asf_tools/composite.py [hunk 1]
```diff
@@ -113,31 +99,24 @@ def reproject_to_median_utm(files, pol, resolution=None):
...
         fi_name = os.path.split(fi)[1]
-        my_zone = get_zone_from_proj(fi)
+        my_epsg_code = get_epsg_code(fi)
         name = fi_name.replace(".tif", "_reproj.tif")
         afi = fi.replace(f"_{pol}.tif", "_area.tif")
         aname = fi_name.replace(f"_{pol}.tif", "_area_reproj.tif")
         if not os.path.isfile(name):
             x, y, trans, proj = saa.read_gdal_file_geo(saa.open_gdal_file(fi))
-            if my_zone != home_zone:
+            if my_epsg_code != target_epsg_code:
```
Comment (asjohnston-asf) [inline asf_tools/composite.py:116]: I suspect this fixes a bug, previously if your input files were say `EPSG:32606` and `EPSG:32706` (same zone different hemisphere) they may not have been projected to a common format.

**Review by jhkennedy — APPROVED** [review]
A decent improvement and paves the way for MGRS if we decide to move that way.

**Comment by asjohnston-asf** [comment]
@talogan Any thoughts?  Did I interpret your strategy correctly in my `get_target_epsg_code` function?

**Comment by asjohnston-asf** [comment]
Regarding testing, I'm thinking the program could:
- marshal the list of input tifs (whether provided via `-i` or via `--path` and `--pol`)
- do a gdalinfo on all the input tifs, which will have all the goodies we need later: projection, pixel size, location on disk, corner coordinates
- then pass around and update the list of gdalinfo dictionaries, so all the business logic functions can be tested simple python objects rather than having to read test data off of disk

**Pull request merged**